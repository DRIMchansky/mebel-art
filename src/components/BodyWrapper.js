// packages
import React from 'react'
import styled from 'styled-components'
// components
import Hamburger from '../components/Hamburger'
import Logo from '../components/Logo'
// utils
import { breakpoints, colors, shadows } from '../util/cssConfig'

const BodyWrapper = ({ children, active, setActive }) => {
  return (
    <BodyWrapperStyled active={active}>
      <MobileHeader>
        <Hamburger active={active} setActive={setActive} />
        <Logo />
      </MobileHeader>
      {children}
    </BodyWrapperStyled>
  )
}
const BodyWrapperStyled = styled.div`
  flex-grow: 1;
  height: 100%;
  display: flex;
  flex-direction: column;
  background: ${colors.white};
  z-index: 2;
  overflow-x: hidden;
  transform: ${(props) => (props.active ? 'translate(60%, 3%)' : 'none')};
  border-radius: ${(props) => (props.active ? '7px' : 'none')};
  transition: all 320ms;
  box-shadow: ${shadows.four};

  @media (min-width: ${breakpoints.medium}) {
  }
`

const MobileHeader = styled.header`
  position: relative;
  padding: 0.9rem 1.2rem;
  text-align: center;

  @media (min-width: ${breakpoints.medium}) {
    display: none;
    visibility: none;
  }
`

export default BodyWrapper
