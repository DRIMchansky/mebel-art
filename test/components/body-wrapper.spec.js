import React from 'react'
import renderer from 'react-test-renderer'
import BodyWrapper from '../../src/components/body-wrapper'

describe('BodyWrapper', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<BodyWrapper />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
