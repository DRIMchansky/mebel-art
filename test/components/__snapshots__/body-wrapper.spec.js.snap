// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BodyWrapper renders correctly 1`] = `
.c2 {
  position: absolute;
  display: inline-block;
  cursor: pointer;
  -webkit-transition-property: opacity,filter;
  transition-property: opacity,filter;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-timing-function: linear;
  transition-timing-function: linear;
  font: inherit;
  color: inherit;
  text-transform: none;
  background-color: transparent;
  border: 0;
  margin: 0;
  overflow: visible;
  left: 10px;
}

.c2:focus {
  outline: none;
}

.c3 {
  width: 40px;
  height: 24px;
  display: inline-block;
  position: relative;
}

.c4 {
  display: block;
  top: 50%;
  margin-top: -2px;
  width: 40px;
  height: 4px;
  background-color: #212529;
  border-radius: 4px;
  position: absolute;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-timing-function: ease;
  transition-timing-function: ease;
}

.c4:before,
.c4:after {
  content: '';
  display: block;
  width: 40px;
  height: 4px;
  background-color: #212529;
  border-radius: 4px;
  position: absolute;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-timing-function: ease;
  transition-timing-function: ease;
}

.c4:before {
  top: -10px;
}

.c4:after {
  bottom: -10px;
}

.c5 {
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  font-weight: 300;
  color: #1c5796;
  font-size: 1.3rem;
  -webkit-letter-spacing: -1px;
  -moz-letter-spacing: -1px;
  -ms-letter-spacing: -1px;
  letter-spacing: -1px;
}

.c5 span {
  font-weight: 800;
  margin-left: 2px;
  color: #283646;
}

.c0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #fffffe;
  z-index: 2;
  overflow: hidden;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  border-radius: none;
  -webkit-transition: all 320ms;
  transition: all 320ms;
  box-shadow: 0px 16px 24px rgba(37,37,37,0.1);
}

.c1 {
  position: relative;
  padding: 0.9rem 1.2rem;
  text-align: center;
}

@media (min-width:600px) {
  .c2 {
    display: none;
    visibility: hidden;
  }
}

@media (min-width:960px) {

}

@media (min-width:600px) {
  .c1 {
    display: none;
    visibility: hidden;
  }
}

<div
  className="c0"
>
  <header
    className="c1"
  >
    <button
      aria-label="Открыть меню"
      className="c2"
      onClick={[Function]}
      type="button"
    >
      <span
        className="c3"
      >
        <span
          className="c4"
        />
      </span>
    </button>
    <a
      className="c5"
      href="/"
    >
      Мебель
      <span>
        Арт
      </span>
    </a>
  </header>
</div>
`;
